qt/applications/workbench/workbench/utils/gather_interfaces.py

```diff
@@ -0,0 +1,85 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2023 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantid workbench.
+
+"""
+Helper functions to collect the names of interfaces
+"""
+
+import os
+from typing import AnyStr, List, Dict
+
+from mantid.kernel import ConfigService, logger
+from mantidqt.usersubwindowfactory import UserSubWindowFactory
+
+
+def get_interface_dir() -> AnyStr:
+    """
+    Returns the path to the directory containing the mantidqt interfaces launch scripts
+    This is the path to the mantidqtinterfaces package
+    """
+    import mantidqtinterfaces
+
+    return os.path.dirname(mantidqtinterfaces.__file__)
+
+
+def _discover_python_interfaces(interface_dir) -> Dict[str, List[str]]:
```

Comment:
Nice idea to move these functions into their own utilities files 👍 

I was reading through this and it reminds me of a similar thing being done on this line of one of our system tests, but more complicated as it is also getting the categories https://github.com/mantidproject/mantid/blob/c1daec527052e9b88866bece07e6fadf35d6030c/Testing/SystemTests/tests/qt/PythonInterfacesStartupTest.py#L42

I was wondering, can you think of a way to simplify this function along the lines of the above L42 ? And then would it be possible to use the new utility function in the `PythonInterfacesStartupTest` too?

Reply:
I can see that I could change it to something like:
```
interfaces = {}
for key, script_name in [interface.split("/") for interface in ConfigService["mantidqt.python_interfaces"].split()]:
    if not os.path.exists(os.path.join(interface_dir, script_name)):
        logger.warning(f'Failed to find script "{script_name}" in "{interface_dir}")
        continue
    interfaces.setdefault(key, []).append(script_name)

return interfaces
```
I want to keep the same dictionary of lists structure and I can't see a simpler way of doing that (like with a dictionary generator) than doesn't use `setdefault()`.